Sources/Workspace/Workspace.swift

```diff
@@ -1885,16 +1887,29 @@ extension Workspace {
                     forceResolution: forceResolution,
                     extraConstraints: extraConstraints,
                     diagnostics: diagnostics,
-                    retryOnPackagePathMismatch: false
+                    retryOnPackagePathMismatch: false,
+                    removeResolvedFile: removeResolvedFile
                 )
-            } else {
+            } else if removeResolvedFile, self.fileSystem.exists(self.resolvedFile) {
                 // If we weren't able to resolve properly even after a retry, it
                 // could mean that the dependency at fault has a different
                 // version of the manifest file which contains dependencies that
                 // have also changed their package references.
-
-                diagnostics.emit(error: "the Package.resolved file is most likely severely out-of-date and is preventing correct resolution; delete the resolved file and try again")
-
+                // TODO: is there a better/safer way to reset the resolved file?
```

Comment:
I've been wondering the same.  @aciidb0mb3r do you know?

Reply:
This should be fine but maybe better to do it via PinsStore object?

```swift
pinsStore.unpinAll()
try pinsStore.saveState()
```